pkg/app/ops/insightcollector/deployments.go

```diff
@@ -0,0 +1,457 @@
+// Copyright 2020 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightcollector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/insight"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+var (
+	pageSize = 50
+)
+
+func (i *InsightCollector) ProcessNewlyCreatedDeployments(ctx context.Context) error {
+	now := time.Now()
+	targetDate := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
+	m, err := i.insightstore.LoadMilestone(ctx)
+	if err != nil {
+		if err == filestore.ErrNotFound {
+			m = &insight.Milestone{}
+		}
+		return err
+	}
+
+	dc, err := i.findDeploymentsCreatedInRange(ctx, m.DeploymentCreatedAtMilestone, targetDate.Unix())
+	if err != nil {
+		return err
+	}
+
+	var updateErr error
+	for _, f := range i.collectFuncs.developmentCollectFns.newlyCreatedDeploymentsFns {
+		if err := f(ctx, dc, targetDate); err != nil {
+			fv := reflect.ValueOf(f)
+			i.logger.Error("failed to execute a collect function",
+				zap.String("function name", runtime.FuncForPC(fv.Pointer()).Name()),
+				zap.Error(err))
+			updateErr = err
+		}
+	}
+
+	if updateErr == nil {
+		m.DeploymentCreatedAtMilestone = targetDate.Unix()
+		if err := i.insightstore.PutMilestone(ctx, m); err != nil {
+			return err
+		}
+	}
+
+	return updateErr
```

Comment:
nit: We always be sure to be deeply nested code so it would be preferred. (L101 as well)
```suggestion
	if updateErr != nil {
		return updateErr
	}
	m.DeploymentCreatedAtMilestone = targetDate.Unix()
	return i.insightstore.PutMilestone(ctx, m)
```